src/common.js

```diff
@@ -4,12 +4,7 @@ let computeFee = (alpha) => {
 }
 
 function calculateFee(value, alpha) {
-  let bigVal = web3.toBigNumber(value)
-  let bigAlpha = web3.toBigNumber(alpha)
-  let bigDenominator = web3.toBigNumber(1000 - alpha)
-  let remainder = bigVal.mul(bigAlpha).modulo(bigDenominator)
-  let result = bigVal.mul(bigAlpha).minus(remainder)
-  return result.div(bigDenominator).add(1)
+  return web3.toBigNumber(value).mul(alpha).div(web3.toBigNumber(1000-alpha)).ceil().toString()
```

Comment:
You are literally the person I was thinking about. Great job. Remove to projection to string though.